### DISCUSSION TITLE
Redis changes the Licence. Maybe KeyDB can be an alternative?

### DISCUSSION THREAD

**Discussion by Kirk1984** [nextcloud/all-in-one#4422] [Ideas] [answered]
https://redis.com/blog/redis-adopts-dual-source-available-licensing/

https://github.com/redis/redis/pull/13157

KeyDB seems to be a drop-in replacement and it's multithreaded:

https://github.com/Snapchat/KeyDB

**Comment by szaimen** [comment]
Hi, thanks for your post! We will consider this but will probably wait until redis 7.2 gets EOL.

**Reply by szaimen** [reply]
https://endoflife.date/redis

**Comment by Zoey2936** [comment]
Just for me, what is the problem on this licence change? Since elasticsearch which is also part of AIO uses the same licence

**Reply by Kirk1984** [reply]
Elasticsearch is optional, redis is not. Plus who knows what the change will bring in the future. For an option it isn't as bad as for a dependency I guess. Maybe I'm jumping a hype train too (a little bit) :)

Plus it seems that KeyDB might be more performant.

**Comment by davidetogni** [comment]
I've no opinion on the future of KeyDB under Snap Inc.
But perhaps valkey (same license, [supported](https://www.linuxfoundation.org/press/linux-foundation-launches-open-source-valkey-community) by the Linux Foudation) could also be considered.

https://github.com/valkey-io/valkey

**Reply by Kirk1984** [reply]
Even better! Didn't know something like that existed 😁

**Reply by RikkiBC** [reply]
Yes please to **Valkey**, @szaimen - Seems the best drop in replacement at least as far as licencing and yeah, backed by Linux Foundation.

**Comment by lastsamurai26** [comment]
What about https://github.com/dragonflydb/dragonfly

**Comment by fulldecent** [comment]
Redix 7.2 EOL is currently Feb 2026

https://redis.io/docs/latest/operate/rs/installing-upgrading/product-lifecycle/

---

Recommending to remove this issue from the priority list https://github.com/nextcloud/all-in-one/issues/5251 because that is a long time from now

**Reply by Zoey2936** [reply]
this is about enterprise, not sure if this also applies to the opensource version AIO uses

**Comment by marktheawesome** [comment]
Valkey seems like a better idea, since its supported by The Linux Foundation. https://github.com/valkey-io/valkey

**Comment by Zoey2936** [comment]
Valkey seems to have an official docker image now: https://hub.docker.com/r/valkey/valkey (not sure since when), it would be a drop-in replacement: https://valkey.io/topics/migration

KeyDB is not possible since it is (currently) based on redis 6.x and since AIO uses redis 7.2.x a downgrade is not possible without losing the cache inside the volumes

**Reply by szaimen** [reply]
If I understand https://endoflife.date/redis correctly, 7.2 security support ends probably around august 2025. Right? So this is the timeframe when we need to change the image latest.

**Reply by Zoey2936** [reply]
not sure, there is no exact date written, if I understand it correctly

**Reply by szaimen** [reply]
you are right. But based on old versions, august 2025 is likely I'd say...

**Reply by Zoey2936** [reply]
yes

**Comment by szaimen** [comment, marked as answer]
Hey, it looks like redis became open source licensed again: https://redis.io/blog/agplv3/

**Reply by szaimen** [reply]
Linking to https://github.com/nextcloud/all-in-one/issues/6387
